src/backend/core/api/viewsets.py

```diff
@@ -156,8 +157,21 @@ def get_queryset(self):
 
             # Filter users by email similarity
             if query := self.request.GET.get("q", ""):
+                # For performance reasons we filter first by similarity, which relies on an index,
+                # then only calculate precise similarity scores for sorting purposes
                 queryset = queryset.filter(email__trigram_word_similar=query)
 
+                queryset = queryset.annotate(
+                    similarity=TrigramSimilarity("email", query)
+                )
+                # When the query only is on the name part, we should try to make many proposals
+                # But when the query looks like an email we should only propose serious matches
+                threshold = 0.6 if "@" in query else 0.1
+
+                queryset = queryset.filter(similarity__gt=threshold).order_by(
+                    "-similarity"
+                )
```

Comment:
Note: I think this was the missing part in the previous code 👍 This should really help not displaying totally wrong email addresses. Also, maybe the `0.1` value is a bit low, but we will adapt it later.